packages/core/tests/index.test.ts

```diff
@@ -472,6 +507,59 @@ describe('@reatom/core', () => {
     })
   })
 
+  test('IoC example', () => {
+    class Api {}
+    const api = new Api()
+    const apiAtom = declareAtom(Symbol('API'), api, () => [])
+
+    var store = createStore(apiAtom)
+    expect(store.getState()).toEqual({
+      [getTree(apiAtom).id]: api,
+    })
+
+    var store = createStore({ [getTree(apiAtom).id]: api })
+    expect(store.getState(apiAtom)).toBe(api)
+    expect(JSON.stringify(store.getState())).toBe('{}')
+  })
+
+  test('createStore replace state', () => {
+    const increment = declareAction()
+    const countAtom = declareAtom(0, on => [on(increment, state => state + 1)])
+    const listener = jest.fn()
+    const store = createStore(countAtom)
+
+    store.subscribe(countAtom, listener)
+
+    expect(store.getState(countAtom)).toBe(0)
+
+    store.dispatch(increment())
+    store.dispatch(increment())
+    const state = store.getState()
+
+    expect(store.getState(countAtom)).toBe(2)
+    expect(listener).toHaveBeenLastCalledWith(2)
+
+    store.dispatch(increment())
+    store.dispatch(increment())
+    expect(store.getState(countAtom)).toBe(4)
+    expect(listener).toHaveBeenLastCalledWith(4)
+
+    store.dispatch({ ...initAction, payload: state })
```

Comment:
Hmm... It seems a bit non-obvious, since "init" occurs twice in this way. Should I use a different name for the action? Or use a special API...